src/test/kotlin/no/nav/familie/ba/sak/behandling/vilkår/VilkårsvurderingStegUtilsTest.kt

```diff
@@ -253,4 +258,65 @@ class VilkårsvurderingStegUtilsTest {
         assertEquals(LocalDate.of(2020, 1, 1), vilkårResultat.periodeFom)
         assertEquals(LocalDate.of(2020, 5, 31), vilkårResultat.periodeTom)
     }
+
+    @Test
+    fun `flyttResultaterTilInitielt filtrer ikke oppfylt om oppfylt finnes`() {
+        val søker = randomFnr()
+        val behandling = lagBehandling()
+
+        val initialVilkårvurdering = lagVilkårsvurderingMedForskelligeResultat(søker, behandling, listOf(Resultat.OPPFYLT))
+        val activeVilkårvurdering = lagVilkårsvurderingMedForskelligeResultat(søker, behandling, listOf(Resultat.IKKE_OPPFYLT, Resultat.OPPFYLT))
+
+        val (initial, active) = VilkårsvurderingUtils.flyttResultaterTilInitielt(initiellVilkårsvurdering = initialVilkårvurdering, aktivVilkårsvurdering = activeVilkårvurdering)
+
+        val opprettetBosattIRiket =
+                initial.personResultater.flatMap { it.vilkårResultater }.filter { it.vilkårType == Vilkår.BOSATT_I_RIKET }
+
+        assertEquals(1, opprettetBosattIRiket.size)
+        assertEquals(Resultat.OPPFYLT, opprettetBosattIRiket.first().resultat)
+    }
+
+    @Test
+    fun `flyttResultaterTilInitielt filtrer ikke ikke oppfylt om oppfylt ikke finnes`() {
+        val søker = randomFnr()
+        val behandling = lagBehandling()
+
+        val initialVilkårvurdering = lagVilkårsvurderingMedForskelligeResultat(søker, behandling, listOf(Resultat.OPPFYLT))
+        val activeVilkårvurdering = lagVilkårsvurderingMedForskelligeResultat(søker, behandling, listOf(Resultat.IKKE_OPPFYLT, Resultat.IKKE_OPPFYLT))
+
+        val (initial, active) = VilkårsvurderingUtils.flyttResultaterTilInitielt(initiellVilkårsvurdering = initialVilkårvurdering, aktivVilkårsvurdering = activeVilkårvurdering)
+
+        val opprettetBosattIRiket =
+                initial.personResultater.flatMap { it.vilkårResultater }.filter { it.vilkårType == Vilkår.BOSATT_I_RIKET }
+
+        assertEquals(2, opprettetBosattIRiket.size)
+        assertFalse(opprettetBosattIRiket.any { it.resultat == Resultat.OPPFYLT })
```

Comment:
I would suggest:
assertTrue(opprettBosattIRiket.all{it.resultat == Resultat.IKKE_OPPFYLT})
because if it.resultat == Resultat.IKKE_REVURDERT, it still can pass the test, but it is not correct

Reply:
What about something like this?

```suggestion
        assertTrue(opprettetBosattIRiket.none { it.resultat == Resultat.OPPFYLT })
```